common/djangoapps/third_party_auth/colaraz.py

```diff
@@ -0,0 +1,60 @@
+from social_core.exceptions import AuthException
+from django.utils.functional import cached_property
+
+from openedx.core.djangoapps.theming.helpers import get_current_request
+from third_party_auth.identityserver3 import IdentityServer3
+
+class ColarazIdentityServer(IdentityServer3):
+    """
+    An extension of the IdentityServer3 for use with Colaraz's IdP service.
+    """
+    name = "colarazIdentityServer"
+    DEFAULT_SCOPE = ["openid", "profile", "IdentityServerApi"]
+    ID_KEY = "email"
+
+    def get_redirect_uri(self, state=None):
+        """
+        Returns redirect uri for oauth redirection
+        """
+        current_req = get_current_request()
+
+        environ = getattr(current_req, "environ", {})
+        schema = environ.get("wsgi.url_scheme", "http")
+
+        site = getattr(current_req, "site", None)
+        domain = getattr(site, "domain", None)
+
+        if not domain:
+            raise AuthException("Colaraz", "Error while authentication")
```

Comment:
Can we use some more descriptive and specific error log here? Mentioning that the error is due to the missing domain.

Reply:
The message is meant to be shown to the user, on his login page. Other than that I have added some logs, will revisit again to make the logs more appropriate.